pkg/kv/followerreads/followerreads_test: TestBoundedStalenessDataDriven failed

pkg/kv/followerreads/followerreads_test.TestBoundedStalenessDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329050?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329050?buildTab=artifacts#/) on master @ [3399ae7c9d6e7b7c840a2355f9fbc50a4cf28ffe](https://github.com/cockroachdb/cockroach/commits/3399ae7c9d6e7b7c840a2355f9fbc50a4cf28ffe):

Failed with:

```
=== RUN   TestBoundedStalenessDataDriven
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven65292266
    test_log_scope.go:82: use -show-logs to present logs inline
=== RUN   TestBoundedStalenessDataDriven/single_row
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
*
* WARNING: disk slowness detected: unable to sync log files within 10s
*
```
Fatal entries found in Cockroach logs:

```
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1  disk stall detected: unable to sync log files within 20s
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !goroutine 159 [running]:
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !runtime/debug.Stack()
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	GOROOT/src/runtime/debug/stack.go:26 +0x64
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !github.com/cockroachdb/cockroach/pkg/util/debugutil.Stack(...)
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	pkg/util/debugutil/debugutil.go:63
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(_, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}}, ...})
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	pkg/util/log/clog.go:292 +0xc4
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x59b4260, 0xa2a6ba0}, 0x2, 0x4, 0x1, 0x1?, {0x563de1b, 0x37}, {0x57752a423fb8, 0x1, ...})
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	pkg/util/log/channels.go:104 +0x484
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !github.com/cockroachdb/cockroach/pkg/util/log.shoutfDepth(...)
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	pkg/util/log/channels.go:111
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !github.com/cockroachdb/cockroach/pkg/util/log.loggerOps.Shoutf(...)
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	bazel-out/aarch64-fastbuild/bin/pkg/util/log/log_channels_generated.go:1072
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).flushAndMaybeSyncLocked.func1()
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	pkg/util/log/file.go:270 +0xa0
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !created by time.goFunc
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !	GOROOT/src/time/sleep.go:215 +0x38
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !For more context, check log files in: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven65292266
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !****************************************************************************
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !This node experienced a fatal error (printed above), and as a result the
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !process is terminating.
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !Fatal errors can occur due to faulty hardware (disks, memory, clocks) or a
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !problem in CockroachDB. With your help, the support team at Cockroach Labs
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !will try to determine the root cause, recommend next steps, and we can
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !improve CockroachDB based on your report.
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !Please submit a crash report by following the instructions here:
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !    https://github.com/cockroachdb/cockroach/issues/new/choose
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !If you would rather not post publicly, please contact us directly at:
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !    support@cockroachlabs.com
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !
F260428 02:03:59.164858 159 1@util/log/file.go:270  [-] 1 !The Cockroach Labs team appreciates your feedback.
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63326

pkg/kv/followerreads/followerreads_test.TestBoundedStalenessDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337748?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337748?buildTab=artifacts#/) on master @ [9a4e05584583e85915a5f865e1aa05d98b46889b](https://github.com/cockroachdb/cockroach/commits/9a4e05584583e85915a5f865e1aa05d98b46889b):


```
=== RUN   TestBoundedStalenessDataDriven
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven3244736266
    test_log_scope.go:82: use -show-logs to present logs inline
    boundedstaleness_test.go:414: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven3244736266
--- FAIL: TestBoundedStalenessDataDriven (6.24s)
=== RUN   TestBoundedStalenessDataDriven/single_row
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20964/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:5:
          exec-system-tenant
          SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s';
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20964/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:9:
          exec
          CREATE TABLE t(pk INT PRIMARY KEY) WITH (schema_locked = false);
          INSERT INTO t VALUES (1);
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20964/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:15:
          query ignore-events
          SELECT DISTINCT(lease_holder) FROM [SHOW RANGES FROM TABLE t WITH DETAILS];
        
        expected:
          1
        
        found:
          2
--- FAIL: TestBoundedStalenessDataDriven/single_row (6.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/kv/followerreads/followerreads_test.TestBoundedStalenessDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21338068?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21338068?buildTab=artifacts#/) on master @ [7c25eaa9935f9d881adf62a5388c044598991839](https://github.com/cockroachdb/cockroach/commits/7c25eaa9935f9d881adf62a5388c044598991839):


```
=== RUN   TestBoundedStalenessDataDriven
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven3200775966
    test_log_scope.go:82: use -show-logs to present logs inline
    boundedstaleness_test.go:414: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven3200775966
--- FAIL: TestBoundedStalenessDataDriven (11.50s)
=== RUN   TestBoundedStalenessDataDriven/single_row
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18973/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:5:
          exec-system-tenant
          SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s';
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18973/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:9:
          exec
          CREATE TABLE t(pk INT PRIMARY KEY) WITH (schema_locked = false);
          INSERT INTO t VALUES (1);
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18973/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:15:
          query ignore-events
          SELECT DISTINCT(lease_holder) FROM [SHOW RANGES FROM TABLE t WITH DETAILS];
        
        expected:
          1
        
        found:
          2
--- FAIL: TestBoundedStalenessDataDriven/single_row (11.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/kv/followerreads/followerreads_test.TestBoundedStalenessDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21339982?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21339982?buildTab=artifacts#/) on master @ [8ac17db7d37daea59b59db93ad77f253249ef65d](https://github.com/cockroachdb/cockroach/commits/8ac17db7d37daea59b59db93ad77f253249ef65d):


```
=== RUN   TestBoundedStalenessDataDriven
I260506 21:41:47.418289 1 (gostd) rand.go:288  [-] 1  random seed: -8295754512354953893
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven3825838218
    test_log_scope.go:82: use -show-logs to present logs inline
    boundedstaleness_test.go:414: -- test log scope end --
--- FAIL: TestBoundedStalenessDataDriven (10.10s)
=== RUN   TestBoundedStalenessDataDriven/single_row
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20047/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:5:
          exec-system-tenant
          SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s';
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20047/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:9:
          exec
          CREATE TABLE t(pk INT PRIMARY KEY) WITH (schema_locked = false);
          INSERT INTO t VALUES (1);
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20047/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:15:
          query ignore-events
          SELECT DISTINCT(lease_holder) FROM [SHOW RANGES FROM TABLE t WITH DETAILS];
        
        expected:
          1
        
        found:
          3
--- FAIL: TestBoundedStalenessDataDriven/single_row (10.07s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/kv/followerreads/followerreads_test.TestBoundedStalenessDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21340745?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21340745?buildTab=artifacts#/) on master @ [c2951c769898e80237c07db3ff516eacac12cf85](https://github.com/cockroachdb/cockroach/commits/c2951c769898e80237c07db3ff516eacac12cf85):


```
=== RUN   TestBoundedStalenessDataDriven
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven4255370819
    test_log_scope.go:82: use -show-logs to present logs inline
    boundedstaleness_test.go:414: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven4255370819
--- FAIL: TestBoundedStalenessDataDriven (6.03s)
=== RUN   TestBoundedStalenessDataDriven/single_row
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10006/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:5:
          exec-system-tenant
          SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s';
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10006/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:9:
          exec
          CREATE TABLE t(pk INT PRIMARY KEY) WITH (schema_locked = false);
          INSERT INTO t VALUES (1);
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10006/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:15:
          query ignore-events
          SELECT DISTINCT(lease_holder) FROM [SHOW RANGES FROM TABLE t WITH DETAILS];
        
        expected:
          1
        
        found:
          3
--- FAIL: TestBoundedStalenessDataDriven/single_row (6.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/kv/followerreads/followerreads_test.TestBoundedStalenessDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21340886?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21340886?buildTab=artifacts#/) on master @ [fa522918a3db0de36c0a90887501c8e4e3fd7777](https://github.com/cockroachdb/cockroach/commits/fa522918a3db0de36c0a90887501c8e4e3fd7777):


```
=== RUN   TestBoundedStalenessDataDriven
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven988134079
    test_log_scope.go:82: use -show-logs to present logs inline
    boundedstaleness_test.go:414: -- test log scope end --
--- FAIL: TestBoundedStalenessDataDriven (10.87s)
=== RUN   TestBoundedStalenessDataDriven/single_row
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19271/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:5:
          exec-system-tenant
          SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s';
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19271/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:9:
          exec
          CREATE TABLE t(pk INT PRIMARY KEY) WITH (schema_locked = false);
          INSERT INTO t VALUES (1);
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19271/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:15:
          query ignore-events
          SELECT DISTINCT(lease_holder) FROM [SHOW RANGES FROM TABLE t WITH DETAILS];
        
        expected:
          1
        
        found:
          3
--- FAIL: TestBoundedStalenessDataDriven/single_row (10.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/kv/followerreads/followerreads_test.TestBoundedStalenessDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21341343?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21341343?buildTab=artifacts#/) on master @ [fcdbe60aa741cba25090ad82250b4b015d2c5ed2](https://github.com/cockroachdb/cockroach/commits/fcdbe60aa741cba25090ad82250b4b015d2c5ed2):


```
=== RUN   TestBoundedStalenessDataDriven
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven779268044
    test_log_scope.go:82: use -show-logs to present logs inline
    boundedstaleness_test.go:414: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b4c6b77bb1c4afd2098de9e97c144fb/logTestBoundedStalenessDataDriven779268044
--- FAIL: TestBoundedStalenessDataDriven (6.01s)
=== RUN   TestBoundedStalenessDataDriven/single_row
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8789/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:5:
          exec-system-tenant
          SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s';
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8789/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:9:
          exec
          CREATE TABLE t(pk INT PRIMARY KEY) WITH (schema_locked = false);
          INSERT INTO t VALUES (1);
          ----
    boundedstaleness_test.go:281: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8789/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/followerreads/followerreads_test_/followerreads_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/followerreads/testdata/boundedstaleness/single_row:15:
          query ignore-events
          SELECT DISTINCT(lease_holder) FROM [SHOW RANGES FROM TABLE t WITH DETAILS];
        
        expected:
          1
        
        found:
          2
--- FAIL: TestBoundedStalenessDataDriven/single_row (5.99s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/followerreads/boundedstaleness_test.go
+++ b/pkg/kv/followerreads/boundedstaleness_test.go
@@ -17,6 +17,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvbase"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvtestutils"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
@@ -276,6 +277,9 @@ func TestBoundedStalenessDataDriven(t *testing.T) {
 		tc.ToggleLeaseQueues(false)
 		tc.ToggleSplitQueues(false)
 		tc.ToggleReplicateQueues(false)
+		kvserverbase.OverrideLoadBasedRebalancingMode(
+			ctx, &tc.Server(0).SystemLayer().ClusterSettings().SV, kvserverbase.LBRebalancingOff,
+		)
 
 		savedTraceStmt := ""
 		datadriven.RunTest(t, path, func(t *testing.T, d *datadriven.TestData) string {

--- a/pkg/kv/followerreads/testdata/boundedstaleness/single_row
+++ b/pkg/kv/followerreads/testdata/boundedstaleness/single_row
@@ -11,6 +11,12 @@ CREATE TABLE t(pk INT PRIMARY KEY) WITH (schema_locked = false);
 INSERT INTO t VALUES (1);
 ----
 
+# Explicitly place the lease on n1 (node_idx=0). The lease can land on any node
+# after table creation depending on Raft leadership of the parent range.
+exec
+ALTER RANGE RELOCATE LEASE TO 1 FOR SELECT range_id FROM [SHOW RANGES FROM TABLE t];
+----
+
 # The tests assume that the leaseholder is node_idx=0 (n1), assert that here.
 query ignore-events
 SELECT DISTINCT(lease_holder) FROM [SHOW RANGES FROM TABLE t WITH DETAILS];

